working-with-collections/main.js

```diff
@@ -1,6 +1,105 @@
+/* eslint-disable no-prototype-builtins */
 console.log('Lodash is loaded:', typeof _ !== 'undefined');
 
-// var player1 = {
-//   name: 'name1',
-//   hand: []
-// };
+function createCards() {
+  var cards = [];
+  var ranks = ['Ace', 2, 3, 4, 5, 6, 7, 8, 9, 10, 'Jack', 'Queen', 'King'];
+  var suits = ['diamonds', 'clubs', 'hearts', 'spades'];
+  for (var i = 0; i < 52; i++) {
+    var card = {};
+    card.rank = ranks[i % 13];
+    card.suit = suits[Math.floor(i / 13)];
+    cards.push(card);
+  }
+  return cards;
+}
+
+function createPlayers(numPlayers) {
+  var players = [];
+  for (var i = 0; i < numPlayers; i++) {
+    var player = {};
+    player.name = 'Player ' + (i + 1);
+    player.hand = [];
+    players.push(player);
+  }
+  return players;
+}
+
+function shuffle(totalCards) {
+  while (cardIndexes.length <= totalCards) {
+    var cardIndex = Math.floor(Math.random() * 52);
+    if (!cardIndexes.includes(cardIndex)) {
+      cardIndexes.push(cardIndex);
+    }
+  }
```

Comment:
If possible, try to keep your functions "pure". Or at least, kinda pure 😄 . 

https://en.wikipedia.org/wiki/Pure_function

When shuffling, there is no way to get around the "impurity" of randomness, but you _can_ avoid modifying a global variable, and instead return a brand new array from the function. Declare your `cardIndexes` variable in here and return it when it's full of cards.